fsspec/caching.py

```diff
@@ -767,14 +767,39 @@ def _read_cache(self, start, end, start_block_number, end_block_number):
 
 
 caches = {
-    "none": BaseCache,
+    # one custom case
     None: BaseCache,
-    "mmap": MMapCache,
-    "bytes": BytesCache,
-    "readahead": ReadAheadCache,
-    "block": BlockCache,
-    "first": FirstChunkCache,
-    "all": AllBytes,
-    "parts": KnownPartsOfAFile,
-    "background": BackgroundBlockCache,
 }
+
+
+def register_cache(cls, clobber=True):
```

Comment:
Ah, if you meant this new value of clobber's default value then yes, I agree that False is better

Reply:
ok, dropped the last commit, no clobber by default here.  Indeed I think it would be better to not clobber on implementations as well, separate PR it is - let's see what that would break ;)